src/app/beer_garden/scheduler.py

```diff
@@ -4,17 +4,369 @@
 from typing import Dict, List
 
 from apscheduler.triggers.interval import IntervalTrigger as APInterval
+
+from watchdog.observers.polling import PollingObserver as Observer
+from watchdog.events import (
+    PatternMatchingEventHandler,
+    EVENT_TYPE_CREATED,
+    EVENT_TYPE_DELETED,
+    EVENT_TYPE_MOVED,
+    EVENT_TYPE_MODIFIED,
+)
+from watchdog.utils import has_attribute, unicode_paths
+from pathtools.patterns import match_any_paths
+
+from apscheduler.schedulers.background import BackgroundScheduler
+
 from brewtils.models import Event, Events, Job
 
 import beer_garden
 import beer_garden.config as config
 import beer_garden.db.api as db
 from beer_garden.events import publish_event
 from beer_garden.requests import process_request, get_request
+from brewtils.models import FileTrigger
+
 
 logger = logging.getLogger(__name__)
 
 
+class InjectionDict(dict):
+    """
+    Dictionary object with overloaded __missing__ function to facilitate partial string.format operations
+    """
+
+    def __missing__(self, key):
+        return "{" + key + "}"
+
+
+def build_injection_dict(dictionary, obj, prefix="", separator="/"):
+    """Populate a dictionary with class variables of an object
+
+    Args:
+        dictionary: A dict-like object to fill
+        obj: An object with some number of variables (e.g MyObj.my_var)
+        prefix: A string to prepend the variable names with
+        separator: A string to separate the prefix and variable names
+
+    """
+    for item in dir(obj):
+        if not callable(getattr(obj, item)):
+            if prefix != "":
+                dictionary[prefix + separator + item] = getattr(obj, item)
+            else:
+                dictionary[item] = getattr(obj, item)
+
+
+def inject_values(request, dictionary):
+    """Inject values into a request
+
+    inject_values looks for string fields and attempts to format() them with the dictionary provided.
+
+    Args:
+        request: An object that may hold string fields with valid str.format() syntax
+        dictionary: A dict-like object to pass through to the format() call
+    """
+    if isinstance(request, dict):
+        for k, v in request.items():
+            try:
+                request[k] = inject_values(v, dictionary)
+            except (ReferenceError, IndexError):
+                pass
+        return request
+
+    elif isinstance(request, str):
```

Comment:
A way that we do this in other places is to use `six.string_types` instead of `str`. An example of this can be seen when we validate request parameters.